DatadogSessionReplay/Sources/Recorder/TouchSnapshotProducer/WindowTouchSnapshotProducer.swift

```diff
@@ -14,12 +14,20 @@ internal class WindowTouchSnapshotProducer: TouchSnapshotProducer, UIEventHandle
     private let windowObserver: AppWindowObserver
     /// Generates persisted IDs for `UITouch` objects.
     private let idsGenerator = TouchIdentifierGenerator()
+    /// Global touch privacy setting
+    private var globalTouchPrivacy: TouchPrivacyLevel
+    /// Keeps track of the privacy override for each touch event
+    private var overrideForTouch = [TouchIdentifier: TouchPrivacyLevel?]()
 
     /// Touches recorded since last call to `takeSnapshot()`
     private var buffer: [TouchSnapshot.Touch] = []
 
-    init(windowObserver: AppWindowObserver) {
+    init(
+        windowObserver: AppWindowObserver,
+        globalTouchPrivacy: TouchPrivacyLevel
+    ) {
         self.windowObserver = windowObserver
+        self.globalTouchPrivacy = globalTouchPrivacy
     }
 
     func takeSnapshot(context: Recorder.Context) -> TouchSnapshot? {
```

Comment:
/suggestion If we want to follow the design in place, we should instead add the `globalTouchPrivacy` as a property of the `Recorder.Context`. The override value could be added to the `TouchSnapshot.Touch` when receiving touches. Then the `takeSnapshot` method will be about getting the first touch that satisfies the privacy level. Would that make sense?

Reply:
Yes, it makes sense! I wish I had thought of that. And actually, the `globalTouchPrivacy` is already a part of [Recorder.Context](https://github.com/DataDog/dd-sdk-ios/blob/2d62ca880be7cca0cb03a30c13d95aad37744cf4/DatadogSessionReplay/Sources/Recorder/Recorder.swift#L30).